android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -1029,24 +1030,33 @@ public TiViewProxy getParent()
 	public String getBackgroundColor()
 	// clang - format on
 	{
-		// Return property if available.
-		if (getProperties().containsKeyAndNotNull(TiC.PROPERTY_BACKGROUND_COLOR)) {
-			return getProperties().getString(TiC.PROPERTY_BACKGROUND_COLOR);
+		// Return originally assigned property value, if available.
+		if (hasPropertyAndNotNull(TiC.PROPERTY_BACKGROUND_COLOR)) {
+			return TiConvert.toString(getProperty(TiC.PROPERTY_BACKGROUND_COLOR));
 		}
+
+		// Fetch the view.
+		TiUIView tiView = getOrCreateView();
+		if (view == null) {
```

Comment:
I suppose the check here should be for `tiView`?

Reply:
Oh wow. Yes, good catch.

Member variable `view` happens to be the return value of `getOrCreateView()`, which is why this happened to work.